src/middlewared/middlewared/plugins/filesystem.py

```diff
@@ -302,6 +297,132 @@ def acl_is_trivial(self, path):
+        key is absent or None, then existing value on the file is not
+        changed.
+
+        `recursive` performs action recursively, but does
+        not traverse filesystem mount points.
+
+        If `traverse` and `recursive` are specified, then the chown
+        operation will traverse filesystem mount points.
+        """
+        self.logger.debug(f'chown: {data}')
+        uid = -1 if data['uid'] is None else data['uid']
+        gid = -1 if data['gid'] is None else data['gid']
+        options = data.get('options')
+
+        if not options['recursive']:
+            os.chown(data['path'], uid, gid)
+        else:
+            winacl = subprocess.run([
+                '/usr/local/bin/winacl',
+                '-a', 'chown',
+                '-O', str(uid), '-G', str(gid),
+                '-rx' if options['traverse'] else '-r',
+                '-p', data['path']], check=False, capture_output=True
+            )
+            if winacl.returncode != 0:
+                raise CallError(f"Failed to recursively change ownership: {winacl.stderr.decode()}")
+
+        self.logger.debug('got here')
+
+    @accepts(
+        Dict(
+            'filesystem_permission',
+            Str('path', required=True),
+            UnixPerm('mode', null=True),
+            Int('uid', null=True, default=None),
+            Int('gid', null=True, default=None),
+            Dict(
+                'options',
+                Bool('stripacl', default=False),
+                Bool('recursive', default=False),
+                Bool('traverse', default=False),
+            )
+        )
+    )
+    @job(lock=lambda args: f'setperm:{args[0]}')
+    def setperm(self, job, data):
+        """
+        Remove extended ACL from specified path.
+
+        If `mode` is specified then the mode will be applied to the
+        path and files and subdirectories depending on which `options` are
+        selected. Mode should be formatted as string representation of octal
+        permissions bits.
+
+        `stripacl` setperm will fail if an extended ACL is present on `path`,
+        unless `stripacl` is set to True.
+
+        `recursive` remove ACLs recursively, but do not traverse dataset
+        boundaries.
+
+        `traverse` remove ACLs from child datasets.
+
+        If no `mode` is set, and `stripacl` is True, then non-trivial ACLs
+        will be converted to trivial ACLs. An ACL is trivial if it can be
+        expressed as a file mode without losing any access rules.
+
+        """
+        options = data.get('options')
+        mode = data.get('mode', None)
+
+        uid = -1 if data['uid'] is None else data['uid']
+        gid = -1 if data['gid'] is None else data['gid']
+
+        if not os.path.exists(data['path']):
+            raise CallError('Path not found.', errno.ENOENT)
+
+        acl_is_trivial = self.middleware.call_sync('filesystem.acl_is_trivial', data['path'])
+        if not acl_is_trivial and not options.get('stripacl', False):
```

Comment:
Default value is set in schema, in cases like this we can use just `options['stripacl']`

Reply:
Thanks. Will correct similar problems.